### PR TITLE
jq 1.7.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Destination channel:** main

### Links

- [PKG-5971](https://anaconda.atlassian.net/browse/PKG-5971) 
- [Upstream repository](https://github.com/jqlang/jq/tree/jq-1.7.1)
- Requirements: 
  - https://github.com/jqlang/jq/blob/jq-1.7.1/configure.ac
  - https://github.com/jqlang/jq/blob/jq-1.7.1/.github/workflows/ci.yml#L68-L171

### Explanation of changes:

- Replace the recipe with the conda-forge one.
- Skip `win` because the `m2-` and `m2w64-` stuff will be replaced by `msys2-` and `ucrt64-` which are currently unavailable. `m2w64-` runtime packages can't be installed with `ucrt64-` in the same environment. See https://anaconda.atlassian.net/browse/PKG-2776?focusedCommentId=241679 and https://anaconda.atlassian.net/browse/PKG-4797. Basically, the upstream source of jq doesn't support Windows without touching msys2 stuff. See also https://anaconda.atlassian.net/browse/PKG-5971?focusedCommentId=244276
- Add `export CFLAGS="-O2 -pthread -fstack-protector-all ${CFLAGS}"` to build.sh, see https://github.com/jqlang/jq/blob/jq-1.7.1/.github/workflows/ci.yml#L162, https://github.com/jqlang/jq/issues/1514, and https://developers.redhat.com/articles/2022/06/02/use-compiler-flags-stack-protection-gcc-and-clang#stack_usage_and_statistics. By default our CFLAGS has `-fstack-protector-strong`
- Add more test commands.

### Notes:

- Currently, `anaconda 2024.10` relies on **jq 1.6** on `win-64`. Disabling the Windows support now will allow us to release the MSYS2/UCRT64 toolchain softly.

[PKG-5971]: https://anaconda.atlassian.net/browse/PKG-5971?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
JC suggested that we could build with `autotools_clang_conda` like we did with `ffmpeg` on win-64. I'm trying it now

**Comment by SC426** [comment]
LGTM! Can you tell me why the patch needed to be added? I don't see any info on it.

**Review by SC426 — APPROVED** [review]

**Comment by skupr-anaconda** [comment]
Skip `win`

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,53 +1,56 @@
...
-      host:
-        - oniguruma 6.9.*  # [not win]
-      run:
-        - oniguruma 6.9.*  # [not win]
-        - m2w64-libwinpthread-git  # [win]
+requirements:
+  build:
+    - gnuconfig  # [unix]
+    - {{ compiler('c') }}
+    - make
```
Comment (cbouss) [inline recipe/meta.yaml:23]:
```suggestion
    - make
    - patch # [linux and (x86_64 or s390x)]
```
To please the linter

**Review by cbouss — APPROVED** [review]
LGTM, with one small suggestion

**Pull request merged**